src/amperity/gocd/secret/vault/plugin.clj

```diff
@@ -198,6 +198,40 @@
 
 
 ;; ## Secret Usage
+(defn- lookup-paths
```

Comment:
I don't think this has enough complexity or reuse by another function to warrant being its own private function, especially when its creating its own anonymous function.

The reduction of path lookups by putting them into a set is nice. The functionality and implementation choices here are tied very closely with the calling function lookup-secrets.

I think this could be folded into lookup-secrets, where that is written more as a composed threading macro or even a transducer if you want to go for efficiency.

But that could be a follow up item later as polish, this is just a suggestion no need to have it block merging.

Reply:
I refactored it and merged `lookup-paths` into `lookup-secrets`👍  Used a threading macro (and got rid of an unnecessary let that was in an anonymous function). Should be more readable now.

Also, untested perk of new implementation is that it should work with any level of nesting in the secrets now (assuming all nested keys are `keywords`):
```
path
path#key
path#key#nested_key...
```